src/charm.py

```diff
@@ -36,6 +45,32 @@ def __init__(self, framework: ops.Framework) -> None:
             framework: The ops framework.
         """
         super().__init__(framework)
+
+        # Get the hostname from the integration data.
+        certificate_requests = []
+        try:
+            nginx_config = get_nginx_config(self)
+            certificate_requests = _generate_certificate_requests(list(nginx_config.keys()))
+        except IntegrationDataError as err:
+            logger.warning("Issues with integration data: %s", err)
+            # Unable to do anything about the error, therefore continue with setup.
+
+        certificates = TLSCertificatesRequiresV4(
+            charm=self,
+            relationship_name=CERTIFICATE_INTEGRATION_NAME,
+            certificate_requests=certificate_requests,
+            mode=Mode.UNIT,
+            refresh_events=[
+                self.on[CACHE_CONFIG_INTEGRATION_NAME].relation_changed,
+                self.on[CACHE_CONFIG_INTEGRATION_NAME].relation_broken,
+            ],
+        )
+        self.certificates_manager = TLSCertificatesManager(
+            user=nginx_manager.NGINX_USER,
+            certificates_path=nginx_manager.NGINX_CERTIFICATES_PATH,
+            certificates=certificates,
+        )
```

Comment:
I think we might need to rethink this a bit, the interface between the certificates manager and this module is a bit too broad here. Let's setup a call to discuss